commercial/app/controllers/BookOffersController.scala

```diff
@@ -20,46 +17,35 @@ class BookOffersController(bookFinder: BookFinder, bestsellersAgent: Bestsellers
   with implicits.Collections
   with implicits.Requests {
 
-  private def booksSample(isbns: Seq[String], segment: Segment): Future[Seq[Book]] =
-    bestsellersAgent.getSpecificBooks(isbns) map { specificBooks =>
-      (specificBooks ++ bestsellersAgent.bestsellersTargetedAt(segment)).distinctBy(_.isbn).take(4)
-    }
+  private def booksSample(isbns: Seq[String], segment: Segment): Seq[Book] =
+    (bestsellersAgent.getSpecificBooks(isbns) ++ bestsellersAgent.bestsellersTargetedAt(segment)).distinctBy(_.isbn).take(4)
+
+  private def isValidIsbn(isbn: String): Boolean = (isbn forall (_.isDigit)) && (isbn.length == 10 || isbn.length == 13)
```

Comment:
Fine. I personally don't like the space notation but as you said you stay consistent in this file. Codebase-wise is another matter though 😋 )
Anyway keep it this way.